### PR TITLE
Fix sorting of nodes by timestamp

### PR DESCRIPTION
**Pull request by adambalogh**

### CODE REVIEW

File: internal/extender/resource.go [hunk 1]
```diff
@@ -196,24 +196,27 @@ func (s *SparkSchedulerExtender) fitEarlierDrivers(
 	return true
 }
 
-func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v1.Pod, nodeNames []string) (string, string, error) {
+func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v1.Pod, nodeNames set) (string, string, error) {
```
Comment (onursatici) [inline internal/extender/resource.go]: I think we should preserve the signature and inline the convertToSet method below

File: internal/extender/resource.go [hunk 2]
```diff
@@ -456,3 +459,13 @@ func (s *SparkSchedulerExtender) deleteResourceReservation(namespace, name strin
 	background := metav1.DeletePropagationBackground
 	return s.resourceReservationClient.ResourceReservations(namespace).Delete(name, &metav1.DeleteOptions{PropagationPolicy: &background})
 }
+
+type set map[string]interface{}
```
Comment (onursatici) [inline internal/extender/resource.go]: I think we use `map[string]bool` in other places, so we can use that for now, and no need to declare a new type as it is not re-used and is descriptive enough

File: internal/extender/resource.go (hunk 1 above)
Comment (adambalogh) [reply internal/extender/resource.go]: makes sense, I actually just extracted this into its own method, can you take a look again?

File: internal/extender/resource.go [hunk 3]
```diff
@@ -198,22 +198,12 @@ func (s *SparkSchedulerExtender) fitEarlierDrivers(
...
 	availableNodes, err := s.nodeLister.List(labels.Set(driver.Spec.NodeSelector).AsSelector())
-	executorNodeNames := make([]string, 0, len(availableNodes))
-	for _, node := range availableNodes {
-		if !node.Spec.Unschedulable {
-			executorNodeNames = append(executorNodeNames, node.Name)
-		}
-	}
 	if err != nil {
-		return "", failureInternal, err
+		return "", failureFit, err
```
Comment (onursatici) [inline internal/extender/resource.go]: should be `failureInternal`

Comment (adambalogh) [reply internal/extender/resource.go]: ops, don't know why I changed that

**Pull request merged**